cypress/integration/ProcessorTest.ts

```diff
@@ -462,22 +473,28 @@ onlyOn(isEnvironmentType(EnvType.Mocked), () => {
       cy.ouiaId("transformation-section", "PF4/Text").should("not.exist");
     }
 
-    function assertFilters(filters: string[][]) {
-      const filtersHeader: string[] = ["Key", "Type", "Value"];
-
+    function assertFilters(
+      filters: { [key in "Key" | "Type" | "Value"]: string }[]
+    ) {
       cy.ouiaId("filters-section", "PF4/Text").should("have.text", "Filters");
-      if (filters?.length > 0 && filters[0].length > 0) {
```

Comment:
Is there any reason why you have removed this condition? Now we are sure that filters has only three types of keys (Key, Type, Value) but someone could input {Key: "data.name", Value: "John"} in such case the code will throw an expection. Is it the expected behaviour? I understand that the previous structure of code is not ideal.

Reply:
I agree this is a place for future improvement. However I do not want to do it as part of this PR, that is already quite large. If I understand correctly the code, even before my refactoring, we didn't validated filter 'deeply'. To me it seems that `input {Key: "data.name", Value: "John"}` was 'valid'  also before my refactoring.